### PR TITLE
Adds trace id to Stripe.Payout

### PR DESCRIPTION
**Pull request by zamith**
Why:

* Trace id a new field coming to Stripe

This change addresses the need by:

* Adding the optional trace_id field. It is currently not supported in all of the countries, but it'll say it's unsupported in that case.

  Also fixes some things to make it work with newer versions of Elixir and Erlang

### CODE REVIEW

File: lib/stripe/core_resources/payout.ex [hunk 1]
```diff
@@ -29,7 +29,8 @@ defmodule Stripe.Payout do
           source_type: String.t(),
           statement_descriptor: String.t() | nil,
           status: String.t(),
-          type: String.t()
+          type: String.t(),
+          trace_id: String.t() | nil
```
Comment (remotecom) [inline lib/stripe/core_resources/payout.ex]: Why string? It is an object.

Comment (kameliya-nr) [reply lib/stripe/core_resources/payout.ex]: I used wrong account. This is me.

File: test/support/stripe_mock.ex [hunk 2]
```diff
@@ -53,14 +53,23 @@ defmodule Stripe.StripeMock do
...
 
+  @impl true
+  def handle_info(
+        {:DOWN, os_pid, :process, _ex_pid, _reason},
+        %{os_pid: os_pid, opts: opts, restarting: false} = state
+      ) do
+    {:ok, manager_pid, os_pid} = start_stripe_mock(opts)
+    {:noreply, %{state | manager_pid: manager_pid, os_pid: os_pid, restarting: false}}
+  end
+
```
Comment (kameliya-nr) [inline test/support/stripe_mock.ex]: Why are these changes needed?

Comment (zamith) [reply test/support/stripe_mock.ex]: They are needed to run the tests, but I guess they could be a separate PR to clean up the repo a bit

Comment (kameliya-nr) [reply test/support/stripe_mock.ex]: Yes, please 🙏🏻

File: test/stripe/core_resources/payout_test.exs [hunk 3]
```diff
@@ -2,29 +2,29 @@ defmodule Stripe.PayoutTest do
...
       assert {:ok, %Stripe.Payout{}} = Stripe.Payout.create(params)
       assert_stripe_requested(:post, "/v1/payouts")
     end
   end
 
   describe "retrieve/2" do
-    test "retrieves a card" do
-      assert {:ok, %Stripe.Payout{}} = Stripe.Payout.retrieve("py_123")
+    test "retrieves a payout" do
+      assert {:ok, %Stripe.Payout{}} = Stripe.Payout.retrieve("py_123") |> dbg()
```
Comment (kameliya-nr) [inline test/stripe/core_resources/payout_test.exs]: Shouldn't we remove the dbg?

**Review by kameliya-nr — APPROVED** [review]

**Review by amalbuquerque — DISMISSED** [review]
Testing if I have the approval rights.

Edit: It seems I can approve it :man_dancing:

**Review by kameliya-nr — APPROVED** [review]

**Review by kaminkaa — APPROVED** [review]
